src/cursor/cur_hs.c

```diff
@@ -381,6 +405,10 @@ __curhs_prev_visible(WT_SESSION_IMPL *session, WT_CURSOR_HS *hs_cursor)
         if (F_ISSET(std_cursor, WT_CURSTD_HS_READ_COMMITTED))
             break;
 
+        /* If the we don't have a transaction snapshot, bail out from visibility checks. */
```

Comment:
Go back to my point in our previous discussion. If we are setting the `WT_CURSTD_HS_READ_COMMITTED` flag in many places. We can just set that flag when it is needed, in this case, for reconciliation, and remove the check for `WT_TXN_HAS_SNAPSHOT `.

Reply:
I have replaced this condition with an assert - if someone using the HS cursor doesn't have the snapshot, they should set the appropriate flag.